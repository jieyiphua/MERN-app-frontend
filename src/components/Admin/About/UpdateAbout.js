import React from 'react';
import { Redirect, withRouter, Link } from 'react-router-dom';
import axios from 'axios';

class UpdateAbout extends React.Component {
  state = {
    data: null,
    selectedContent: null,
    description: this.props.selectedContent.description || '',
    id: this.props.selectedContent._id,
    errors: []
  }

  componentDidMount() {
    console.log('About component did mount')
  }

  onInputChange = (event) => {
    this.setState({ description: event.target.value })
    console.log(this.state)
  }

  onFormSubmit = async (event) => {
    event.preventDefault();

    try {
      console.log(this.state.description)
      const response = await axios.put(`https://deployment-mern-backend-tessivanjayz.gardtess.now.sh/about/${this.state.id}`, {description: this.state.description})

      console.log(response)
      console.log(response.config)
    } catch (error) {
      this.setState({
        errors: error.response
      })
    }
    this.props.history.push('/about/our-story')
  }

  render() {
    const { authentication } = this.props

    if (!authentication) {
      return <Redirect to="/admin/login" />
    }
    console.log('about render')
    console.log(this.state)

    return (
      <div style={{position: 'absolute', left: '200px', top: '200px'}}>
        <h1>Update About</h1>
        <form  onSubmit={this.onFormSubmit} style={{display: 'flex', flexDirection: 'column'}}>
          <label htmlFor="description"> Description:</label>
          <textarea rows="10" cols="50" value={this.state.description} onChange={this.onInputChange} name="description" id="description"/>
          <button onClick={this.onFormSubmit}>Submit</button>
          <Link to="/about/our-story"><button>Back</button></Link>
        </form>
      </div>
    )
  }
}

export default withRouter(UpdateAbout);