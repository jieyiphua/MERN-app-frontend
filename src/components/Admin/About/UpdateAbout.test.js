import UpdateAbout from "./UpdateAbout";
import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";

describe("<UpdateAbout />", () => {
  it("renders component", () => {
    const wrapper = shallow(<UpdateAbout />);
    expect(wrapper).toMatchSnapshot();
  });
});
