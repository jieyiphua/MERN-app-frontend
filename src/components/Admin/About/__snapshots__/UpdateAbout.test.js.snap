// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateAbout /> renders component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
